lib/src/internal/data_client.dart

```diff
@@ -52,15 +55,35 @@ class DataClient implements AbstractDataClient {
   late ScsClient _client;
   final CacheClientConfiguration _configuration;
   final Duration _defaultTtl;
+  final MomentoLogger _logger = MomentoLogger('MomentoCacheDataClient');
+  var firstRequest = true;
 
   DataClient(CredentialProvider credentialProvider, this._configuration,
       this._defaultTtl) {
     _channel = ClientChannel(credentialProvider.cacheEndpoint);
     _client = ScsClient(_channel,
-        options: CallOptions(metadata: {
-          'authorization': credentialProvider.apiKey,
-          'agent': 'dart:0.1.0',
-        }, timeout: _configuration.transportStrategy.grpcConfig.deadline));
+        options: CallOptions(
+            metadata: {'authorization': credentialProvider.apiKey},
+            timeout: _configuration.transportStrategy.grpcConfig.deadline));
+  }
+
+  Future<Map<String, String>> makeHeaders({String? cacheName}) async {
+    var headers = <String, String>{};
+    if (cacheName != null) {
+      headers.addEntries({'cache': cacheName}.entries);
+    }
+    if (firstRequest) {
```

Comment:
might be worth pulling this whole clause into your utils namespace? not a blocker.